src/inputs/DateFields/DateFieldBase.tsx

```diff
@@ -223,18 +228,21 @@ export function DateFieldBase(props: DateRangeFieldBaseProps | DateSingleFieldBa
     (d: Date | DateRange | undefined) => {
       setWipValue(d);
       if (d && isParsedDateValid(d)) {
-        if (props.mode === "range" && isDateRange(d)) {
+        if (isRangeMode && isDateRange(d)) {
           props.onChange(d);
           return;
         }
 
-        if (props.mode === "single" && !isDateRange(d)) {
+        if (!isRangeMode && !isDateRange(d)) {
           props.onChange(d);
           return;
         }
+      } else if (isRangeMode && isRangeFilterField) {
+        props.onChange(undefined);
+        return;
       }
```

Comment:
**question/thought:** I'm thinking we might just want to allow DateRangeField to always support `undefined`. Our props have been updated to support `undefined`, but the implementation is avoiding it here unless it is the filter field. I think we might be creating too many code paths by doing it this way, and instead I would prefer if we allow any DateRangeField to return `undefined` (as the props are going to assume that it can happen anyways)

Reply:
Makes sense, i'll get that updated and tested. I can see it causing some issues in our filter definitions but i should be able to work it out.